src/tests/ftest/util/run_utils.py

```diff
@@ -368,3 +369,70 @@ def find_command(source, pattern, depth, other=None):
     elif isinstance(other, str):
         command.append(other)
     return " ".join(command)
+
+
+def stop_processes(log, hosts, pattern, verbose=True, timeout=60, exclude=None, force=False):
+    """Stop the processes on each hosts that match the pattern.
+
+    Args:
+        log (logger): logger for the messages produced by this method
+        hosts (NodeSet): hosts on which to stop any processes matching the pattern
+        pattern (str): regular expression used to find process names to stop
+        verbose (bool, optional): display command output. Defaults to True.
+        timeout (int, optional): command timeout in seconds. Defaults to 60 seconds.
+        exclude (str, optional): negative filter to better identify processes. Defaults to None.
+        force (bool, optional): if set use the KILL signal to immediately stop any running
+            processes. Defaults to False which will attempt to kill w/o a signal, then with the ABRT
+            signal, and finally with the KILL signal.
+
+    Returns:
+        tuple: (NodeSet, NodeSet) where the first NodeSet indicates on which hosts processes
+            matching the pattern were initially detected and the second NodeSet indicates on which
+            hosts the processes matching the pattern are still running (will be empty if every
+            process was killed or no process matching the pattern were found).
+
+    """
+    processes_detected = NodeSet()
+    processes_running = NodeSet()
+    command = f"/usr/bin/pgrep --list-full {pattern}"
+    pattern_match = str(pattern)
+    if exclude:
+        command = f"/usr/bin/ps xa | grep -E {pattern} | grep -vE {exclude}"
```

Comment:
We should probably quote the args to be safe
```suggestion
        command = f"/usr/bin/ps xa | grep -E '{pattern}' | grep -vE '{exclude}'"
```